lib/offline.js

```diff
@@ -63,52 +70,111 @@ module.exports = function(config) {
         return '';
       }
     }).then(function(cacheId) {
-      var staticFileGlobs = fileGlobs.map(function(v) {
+      checkImports(importScripts, rootDir);
+
+      var absoluteGlobs = fileGlobs.map(function (v) {
         return path.join(rootDir, v);
       });
+      var files = flatGlobs(absoluteGlobs);
+      var filesAndHashes = getFilesAndHashes(files, rootDir, absoluteGlobs);
 
-      staticFileGlobs.forEach(function(globPattern) {
-        glob.sync(globPattern.replace(path.sep, '/')).forEach(function(file) {
-          var stat = fs.statSync(file);
+      var replacements = {
+        cacheId: cacheId,
+        ignoreUrlParametersMatching: config.ignoreUrlParametersMatching || [/./], //TODO: Not implemented in template
+        cacheVersion: getHashOfHashes(pluckHashes(filesAndHashes)),
+        resources: filesAndHashes,
+        importScripts: importScripts,
+      };
 
-          if (stat.isFile() && stat.size > 2 * 1024 * 1024 && staticFileGlobs.indexOf(file) === -1) {
-            console.log(chalk.yellow.bold(file + ' is bigger than 2 MiB. Are you sure you want to cache it? To suppress this warning, explicitly include the file in the fileGlobs list.'));
-          }
-        });
+      var stream = gulp.src([__dirname + '/../templates/app/offline-worker.js'])
+        .pipe(template(replacements))
+        .pipe(conflict(rootDir))
+        .pipe(gulp.dest(rootDir));
```

Comment:
Doing this with Gulp has the unfortunate side-effect of displaying a complex and confusing error message at the end of the process:

```
[13:39:15] [conflict] Creating offline-worker.js
```

We shouldn't display this message, per the design proposal in #162 and its implementation in #175. (If we did decide that it made sense to tell the user we were creating that file, then it would look very different, something like `✓ Creating PATH/TO/offline-worker.js'.) Can you suppress that output?